CryptoPrices/Sources/Styleguide/README.md

```diff
@@ -0,0 +1,3 @@
+# Styleguide
+
+Typography, colors, assets, common views, and other styling-related stuff.
```

Comment:
Since it looks like the `Styleguide` module only supports the view's related stuff, how about shared non-view related stuff like public booleans, other constants, device settings, translations, etc? It seems that we will still need to create another shared module, right? So, why don't we turn the `Styleguide` into `Shared` instead for future extendability?

Reply:
Thanks, @minhnimble for the concerns! I'd like to share my idea about how we're going to modularize our application 🙏 
- As its name implies and its description, the `Styleguide` module contains only the view's related stuff. That's because in order to leverage modulization, we have to think carefully about which stuff should go to one or another module. For e.g in case, we need to expose common constants and settings, we should group them into a separate module like `Constants` or `Settings`. It has some great benefits like:
  - Avoid a large, giant shared module. We should try to further divide and package stuff into smaller modules.
  - Reason about how to use from the perspective of the module consumer (importer). Let's say we combine all view and settings stuff into a single module `SharedModule`. So a view-related module like `Home` in order to use just some fonts, colors, and common views has to import all include the data stuff they don't want. On the other hand, it's a little bit strange that a data-related module like `Repository` in order to use some settings, constants like `apiURL` imports a module with view stuff.
  - Easily organize, develop, and maintain.
  - Good if not great build time. If we package (group) related stuff to different, separate modules, the compiler doesn't have to build a single, large module which may take a long time. It can parallelly build multiple modules at the same time. And speaking of code changes, the change on this module won't cause to rebuild of unrelated modules. For e.g. the `Settings` module which contains some constants that rarely change, we almost _set and forget_. But the `Styleguide` module may change more frequently for some reasons like we add a new font style, or changing the alpha of the existing color, ... If we put them all into a single module, any small change like adding new font will cause the whole big module rebuild, include the Swift files that declare some constants.

- About the naming of the modules, I think we should name them based on ___what__ they provide and expose_ rather than ___how__ they are being used_. More clearly is that we shouldn't bother with the `shared` word even if they're being _shared_ in the sense that one or more other parts will import and use them. For e.g. instead of `SharedViews` we should name it `Styleguide`, `DesignSystem`, or `CustomViews`. Instead of `SharedModels`, we can go with `DomainModels` or `Entities` 🙏

Reply:
@Thieurom I think you had valid reasons there, great points 👍 More small modules even for the shared category there should be a great direction as long as we manage the potential duplication well. Let's stick with this approach then. I think we should note this somewhere for later usage 😁